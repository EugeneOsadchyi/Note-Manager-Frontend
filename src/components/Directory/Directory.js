import React from 'react';
import PropTypes from 'prop-types';

const Directoty = ({ name }) => (
  <div>
    <span className="name">
      {name}
    </span>
  </div>
);

Directoty.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Directoty;
