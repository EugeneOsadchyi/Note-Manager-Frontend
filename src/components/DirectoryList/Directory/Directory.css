.Directory.active {
  background-color: blue;
}
