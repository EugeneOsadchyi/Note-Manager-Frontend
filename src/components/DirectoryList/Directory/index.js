import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFolder, faFolderOpen } from '@fortawesome/free-solid-svg-icons';


const Directory = ({
  name,
  opened,
  folderClicked,
  doubleClicked,
}) => {
  const icon = opened ? faFolderOpen : faFolder;

  return (
    <div className="Directory" onDoubleClick={doubleClicked}>
      <span className="icon" onClick={folderClicked}>
        <FontAwesomeIcon icon={icon} />
      </span>
      <span className="name">
        {name}
      </span>
    </div>
  );
};

Directory.propTypes = {
  name: PropTypes.string.isRequired,
  opened: PropTypes.bool.isRequired,
  doubleClicked: PropTypes.func.isRequired,
  folderClicked: PropTypes.func.isRequired,
};

export default Directory;
