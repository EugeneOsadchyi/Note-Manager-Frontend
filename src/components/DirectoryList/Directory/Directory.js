import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFolder, faFolderOpen } from '@fortawesome/free-solid-svg-icons';
import './Directory.css';

const Directory = ({
  name,
  opened,
  active,
  clicked,
  doubleClicked,
  folderClicked,
}) => {
  const icon = opened ? faFolderOpen : faFolder;
  const classes = classNames('Directory', {
    active,
  });

  return (
    <div
      className={classes}
      onClick={clicked}
      onDoubleClick={doubleClicked}
      role="presentation"
    >
      <span className="icon" role="presentation" onClick={folderClicked}>
        <FontAwesomeIcon icon={icon} />
      </span>
      <span className="name">
        {name}
      </span>
    </div>
  );
};

Directory.propTypes = {
  name: PropTypes.string.isRequired,
  opened: PropTypes.bool.isRequired,
  active: PropTypes.bool.isRequired,
  clicked: PropTypes.func.isRequired,
  doubleClicked: PropTypes.func.isRequired,
  folderClicked: PropTypes.func.isRequired,
};

export default Directory;
