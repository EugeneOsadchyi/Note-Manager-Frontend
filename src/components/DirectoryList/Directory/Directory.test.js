import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Directory from './Directory';

describe('Directory', () => {
  const name = 'Test Directory';
  const opened = false;
  const active = false;
  const mockClicked = jest.fn();
  const mockDoubleClicked = jest.fn();
  const mockFolderClicked = jest.fn();

  const props = {
    name,
    opened,
    active,
    clicked: mockClicked,
    doubleClicked: mockDoubleClicked,
    folderClicked: mockFolderClicked,
  };
  let directory = shallow(<Directory {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(directory)).toMatchSnapshot();
  });

  it('displays directory name from the props', () => {
    expect(directory.find('.name').text()).toEqual(name);
  });

  describe('when `opened` is received from the props', () => {
    beforeEach(() => {
      props.opened = true;
      directory = mount(<Directory {...props} />);
    });

    it('displays opened folder icon', () => {
      expect(directory.find('.icon .fa-folder-open').exists()).toBe(true);
    });
  });

  describe('when not `opened` is received from the props', () => {
    beforeEach(() => {
      props.opened = false;
      directory = mount(<Directory {...props} />);
    });

    it('displays closed folder icon', () => {
      expect(directory.find('.icon .fa-folder').exists()).toBe(true);
    });
  });

  describe('when directory name was double-clicked', () => {
    beforeEach(() => directory.find('.name').simulate('doubleclick'));

    it('dispatches the `doubleClicked()` it receives from the props', () => {
      expect(mockDoubleClicked).toHaveBeenCalled();
    });
  });

  describe('when directory icon was clicked', () => {
    beforeEach(() => directory.find('.icon').simulate('click'));

    it('dispatches the `folderClicked()` it receives from the props', () => {
      expect(mockFolderClicked).toHaveBeenCalled();
    });
  });

  describe('when directory was clicked', () => {
    beforeEach(() => directory.simulate('click'));

    it('dispatched the `clicked()` it receives from the props', () => {
      expect(mockClicked).toHaveBeenCalled();
    });
  });

  describe('when `active` is received from the props', () => {
    beforeEach(() => {
      props.active = true;
      directory = shallow(<Directory {...props} />);
    });

    it('renders properly', () => {
      expect(shallowToJson(directory)).toMatchSnapshot();
    });

    it('highlights the directory', () => {
      expect(directory.find('.active').exists()).toBe(true);
    });
  });
});
