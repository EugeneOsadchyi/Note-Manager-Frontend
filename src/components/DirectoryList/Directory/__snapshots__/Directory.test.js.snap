// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory renders properly 1`] = `
<div
  className="Directory"
  onClick={[Function]}
  onDoubleClick={[Function]}
>
  <span
    className="icon"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f07b",
            "M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z",
          ],
          "iconName": "folder",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </span>
  <span
    className="name"
  >
    Test Directory
  </span>
</div>
`;

exports[`Directory when \`active\` is received from the props renders properly 1`] = `
<div
  className="Directory active"
  onClick={[Function]}
  onDoubleClick={[Function]}
>
  <span
    className="icon"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f07b",
            "M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z",
          ],
          "iconName": "folder",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
  </span>
  <span
    className="name"
  >
    Test Directory
  </span>
</div>
`;
